tools/InstallationTests/AutoTestLogin.ps1

```diff
@@ -0,0 +1,490 @@
+    }
+    else
+    {
+        Log-Info "Successfully installed certificate to '$global:gLocalCertStore'"
+    }
+}
+
+Function Download-TestCertificateFromKeyVault()
+{
+    #$certContentType = "application/x-pkcs12"
+    #$certName = "PsAutoTestCertName"
+    #$vaultName = "KV-PsSdkAuto"
+
+    Check-LoggedInContext
+    $kvCertSecret = Get-AzureKeyVaultSecret -VaultName $global:gVaultName -Name $global:kvSecKey_PsAutoTestCertNameKey
+
+    #ToDo: Handle the case where access denied to keyVault will result in an attempt to create a new certificate (which will fail again, but needs to be handled)
+    if($kvCertSecret -eq $null)
+    {
+        Log-Info "Unable to get certificate from Azure KeyVault"
+        $kvCertSecret = Create-TestAutoCertificateInKv $gVaultName $kvSecKey_PsAutoTestCertNameKey
+    }
+    
+    #Once we create certificate and get it, we store it locally
+    if($kvCertSecret -ne $null)
+    {
+        $kvSecretBytes = [System.Convert]::FromBase64String($kvCertSecret.SecretValueText)
+        $certCollection2 = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2Collection
+        $certCollection2.Import($kvSecretBytes, $null, [System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::Exportable)
+        
+        #Save Pfx locally
+        $pwd = Get-LocalCertificatePassword
+        $certPwdProtectedInBytes = $certCollection2.Export([System.Security.Cryptography.X509Certificates.X509ContentType]::Pkcs12, $pwd)
+        $pfxLocalFilePath = [System.IO.Path]::Combine($global:localPfxDirPath, $global:gPfxLocalFileName)
+        [System.IO.File]::WriteAllBytes($pfxLocalFilePath, $certPwdProtectedInBytes)
+
+        #We are not going the .cer route for now, rather use publishSettings file
+        #Save .cer file location to be uploaded to the management portal
+        #$kvAutoTestCert = Get-AzureKeyVaultCertificate -VaultName $vaultName -Name $certName
+        #$cerLocalFilePath = [Environment]::GetFolderPath("Desktop") + "\PsAutoTestCert.cer"
+        #$certBytes = $kvAutoTestCert.Certificate.Export([System.Security.Cryptography.X509Certificates.X509ContentType]::Cert)
+        #[System.IO.File]::WriteAllBytes($cerLocalFilePath, $certBytes)
+
+        Log-Info "Successfully downloaded certificate at '$pfxLocalFilePath'"
+    }
+
+    return $pfxLocalFilePath
+}
+
+Function Get-LocalCertificatePassword()
+{
+    if([string]::IsNullOrEmpty($global:gCertPwd) -eq $true)
+    {
+        $global:gCertPwd = Read-Host "Please enter certificate password that is ready to be installed on your machine"        
+    }
+
+    return $global:gCertPwd
+}
+
+Function Login-AzureRMWithCertificate()
+{
+    $appId = $global:gPsAutoTestADAppId
+
+    $localCert = Get-LocalCertificate
+    if($localCert -eq $null)
+    {
+        #we could not find certificate installed locally, get it from KeyVault
+        Login-InteractivelyAndSelectTestSubscription
```

Comment:
This should be protected by SHouldContinue, so it will just fail if run as part of a ccript.  This means the function should eb a cmdlet